kroxylicious-filters/kroxylicious-record-encryption/src/main/java/io/kroxylicious/filter/encryption/RecordEncryption.java

```diff
@@ -100,25 +105,86 @@ private static void checkCipherSuite() {
+    @Nullable
+    private static Integer getExperimentalInt(Map<String, Object> experimental, String property) {
+        return Optional.ofNullable(experimental).map(x -> x.get(property)).map(value -> {
+            if (value instanceof Number number) {
+                return number.intValue();
+            }
+            else if (value instanceof String stringValue) {
+                return Integer.parseInt(stringValue);
+            }
+            else {
+                throw new IllegalArgumentException("could not convert " + property + " with type " + value.getClass().getSimpleName() + " to Integer");
+            }
+        }).orElse(null);
+    }
+
+    @Nullable
+    private static Long getExperimentalLong(Map<String, Object> experimental, String property) {
+        return Optional.ofNullable(experimental).map(x -> x.get(property)).map(value -> {
+            if (value instanceof Number number) {
+                return number.longValue();
+            }
+            else if (value instanceof String stringValue) {
+                return Long.parseLong(stringValue);
+            }
+            else {
+                throw new IllegalArgumentException("could not convert " + property + " with type " + value.getClass().getSimpleName() + " to Integer");
+            }
+        }).orElse(null);
+    }
+
     @Override
     public SharedEncryptionContext<K, E> initialize(FilterFactoryContext context,
                                                     RecordEncryptionConfig configuration)
             throws PluginConfigurationException {
         checkCipherSuite();
-        KmsService<Object, K, E> kmsPlugin = context.pluginInstance(KmsService.class, configuration.kms());
-        kmsPlugin.initialize(configuration.kmsConfig());
+        var kmsPlugin = context.pluginInstance(KmsService.class, configuration.kms());
+        kmsPlugin.initialize(convertConfigValueToConfigType(kmsPlugin, configuration.kmsConfig()));
         Kms<K, E> kms = buildKms(configuration, kmsPlugin);
 
-        var dekConfig = configuration.dekManager();
+        var dekConfig = dekManager(configuration);
         DekManager<K, E> dekManager = new DekManager<>(kms, dekConfig.maxEncryptionsPerDek());
 
-        KmsCacheConfig cacheConfig = configuration.kmsCache();
+        KmsCacheConfig cacheConfig = kmsCache(configuration);
         EncryptionDekCache<K, E> encryptionDekCache = new EncryptionDekCache<>(dekManager, null, EncryptionDekCache.NO_MAX_CACHE_SIZE,
                 cacheConfig.encryptionDekCacheRefreshAfterWriteDuration(), cacheConfig.encryptionDekCacheExpireAfterWriteDuration());
         DecryptionDekCache<K, E> decryptionDekCache = new DecryptionDekCache<>(dekManager, null, DecryptionDekCache.NO_MAX_CACHE_SIZE);
         return new SharedEncryptionContext<>(kms, kmsPlugin::close, configuration, dekManager, encryptionDekCache, decryptionDekCache);
     }
 
+    private static <P, C> @NonNull C convertConfigValueToConfigType(P plugin, Object conf) {
```

Comment:
This method is a bit of a temporary hack. Really we want the _runtime_ to know about the schemas exposed by each plugin, and how they're reflected as Java types.

Reply:
This temporary hack is no more.